pkg/rootless/rootless_linux.go

```diff
@@ -97,6 +98,13 @@ func BecomeRootInUserNS() (bool, int, error) {
 
 	var uids, gids []idtools.IDMap
 	username := os.Getenv("USER")
+	if username == "" {
+		user, err := user.LookupId(fmt.Sprintf("%d", os.Geteuid()))
```

Comment:
Does this work with usernames not in /etc/passwd?

Reply:
no, it works only with users defined in `/etc/passwd`.

I've changed the error message with a pointer to the USER env variable

Reply:
Do any go interfaces work with nsswitch?

Reply:
from a quick search I didn't find anything, I'll look further if there is something available

Reply:
@vbatts Do you know of anything?

Reply:
@giuseppe @rhatdan You could use `github.com/opencontainers/runc/libcontainer/user` as we do in psgo (https://github.com/containers/psgo/blob/master/internal/process/process.go#L55).

Reply:
@vrothberg looking at the code, it seems it also just parses `/etc/passwd`

Reply:
RIght I think the libcontainer/user code is just for parsing the username inside of the container.

We need code to call into the gcc getpw library so that it will work with ldap and other providers of identity like freeipa.